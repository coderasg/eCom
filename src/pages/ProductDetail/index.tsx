import React from 'react';
import { useParams } from 'react-router-dom';

/**
 * API
 */
import useProductDetail from '../../hooks/fetchAPIs/useProductDetail';

/**
 * Common Components
 */
import Header from '../../components/Header';
import ProductImage from './../../components/ProductImage/index';

/**
 * Page Related Components
 */
import ProductDetailHeader from './components/ProductDetailHeader';
import ProductDescription from './components/ProductDescription';
import ProductReviews from './components/ProductReviews';
import ProceedToCheckout from './components/ProceedToCheckout';

import styles from './productDetail.module.scss';

function ProductDetail() {
  const { id } = useParams();
  const { data: product, error } = useProductDetail(`${id}`);

  if (error) {
    return <h1>{'Failed to load'}</h1>;
  }
  if (!product) {
    return <h1>{'Loadiding..............'}</h1>;
  }

  return (
    <div data-testid="ProductDetail-wrapper" className={styles.productDetail}>
      <Header>
        <ProductDetailHeader title={product.name} />
      </Header>
      <section className={styles.productContent}>
        <ProductImage src={product.image} name={product.name} bannerText={product.is_sold_out ? 'SOLD' : ''} />
        <ProductReviews commentCount={product.comment_count} likeCount={product.like_count} />
        <ProductDescription decription={product.description} />
      </section>
      <ProceedToCheckout price={product.price} shippingFee={product.shipping_fee}  />
    </div>
  );
}

export default ProductDetail;
