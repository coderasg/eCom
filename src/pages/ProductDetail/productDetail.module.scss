
.productContent {
    padding: 5px;
    margin-top: 10px;
    margin-bottom: 60px;
}