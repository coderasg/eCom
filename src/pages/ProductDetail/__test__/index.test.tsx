import React from 'react';
import renderer from 'react-test-renderer';

import ProductDetail from '../index';

describe('(Component) ProductDetail', () => {
  const defaultProps = {};
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<ProductDetail {...defaultProps} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
