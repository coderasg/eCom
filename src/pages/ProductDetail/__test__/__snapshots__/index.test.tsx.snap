// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ProductDetail should pass snapshot test 1`] = `
<h1>
  Failed to load
</h1>
`;
